lib/utils/CountryHelper.dart

```diff
@@ -1004,3 +1004,12 @@ extension OpenFoodFactsCoutryExtension on OpenFoodFactsCountry {
 
   String get iso2Code => _ISO_2_CODES[this]!;
 }
+
+/// Helper class around [OpenFoodFactsCountry]
+class CountryHelper {
+  /// Converts an ISO 2 code into an [OpenFoodFactsCountry]
+  static OpenFoodFactsCountry? fromJson(String? code) =>
+      OpenFoodFactsCountry.values.firstWhere(
+        (final OpenFoodFactsCountry country) => country.iso2Code == code,
+      );
```

Comment:
Does not work for an unknown `code`: crashes, instead of an expected `null`.
`firstWhere` is not adequate, as it never returns null.
Your previous syntax was correct (it's just that the `return;` was useless).

Reply:
Yes I know that it theoretically worked but it didn't make sense since the extension only worked for specific values of the enum, not for the enum itself, so there you would have to write `OpenFoodFactsCountry.FRANCE.fromCountryCode('de')` which would return germany, that's why I moved it and now the structure is similar to the one in the OpenFoodFactsLanguage.

Regarding the null problem, it's not possible to return null from `orElse` so we could either add a UNKNOW value to the enum as in the OpenFoodFactsLanguage but I would rather just switch back to the old forEach loop.

Reply:
I see, `static` methods don't seem to be allowed in `extension`s.

Similarly to what you wrote initially, and which seems to work:
```dart
class CountryHelper {
  static OpenFoodFactsCountry? fromJson(String? code) {
    OpenFoodFactsCountry? result;
    OpenFoodFactsCountry.values.forEach((final OpenFoodFactsCountry key) {
      if (key.iso2Code == code) {
        result = key;
      }
    });
    return result;
  }
}
```

Reply:
Or even better (I don't find the dart map methods always 100% relevant):
```dart
class CountryHelper {
  static OpenFoodFactsCountry? fromJson(String? code) {
    if (code == null) {
      return null;
    }
    for (final OpenFoodFactsCountry key in OpenFoodFactsCountry.values) {
      if (key.iso2Code == code) {
        return key;
      }
    }
    return null;
  }
}
```

Reply:
Changed it, good suggestion